model/src/main/kotlin/Hash.kt

```diff
@@ -78,6 +78,11 @@ data class Hash(
                     "'$value' is not a $algorithm hash."
                 }
             }
+
+        /**
+         * Create a [Hash] instance similar to [create](String,String).
```

Comment:
I'm not aware that we refer to other functions that way, and I'd avoid it to not expose any implementation details, but just describe the `[algorithm]` parameter in the usual way.

Reply:
Also, the question here (even more than for the above overload) is: Why not use the `Hash()` constructor directly. For both cases below, it's clear from the property name that the hash is always a SHA256, so I'd simply call the `Hash()` constructor instead of introducing this overload.

Reply:
With https://github.com/oss-review-toolkit/ort/pull/8905, we maybe could even drop the above original overload and instead verify in an `init` block that `value.length == algorithm.size`.

Reply:
> Also, the question here (even more than for the above overload) is: Why not use the `Hash()` constructor directly. For both cases below, it's clear from the property name that the hash is always a SHA256, so I'd simply call the `Hash()` constructor instead of introducing this overload.

I guess there has to be a reason for these sanity checks, which is why I went for this overload. 
This way it remains a non-functional change as it keeps that sanity check.

Reply:
> I guess there has to be a reason for these sanity checks

I don't think so. Maybe it was just over-cautiousness, or simply an oversight. But I really do not see an issue with using the `Hash()` constructor, esp. if it's `init`  block would introduce basically the same sanity check on the hash string value size.

Reply:
Ok, I'll close this PR then.